src/include/splash/basetypes/ColTypeString.hpp

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright 2013-2015 Felix Schmitt, Axel Huebl
+ *
+ * This file is part of libSplash.
+ *
+ * libSplash is free software: you can redistribute it and/or modify
+ * it under the terms of of either the GNU General Public License or
+ * the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * libSplash is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License and the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * and the GNU Lesser General Public License along with libSplash.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef COLTYPESTRING_H
+#define	COLTYPESTRING_H
+
+#include "splash/CollectionType.hpp"
+
+namespace splash
+{
+    /** Variable and fixed length strings
+     *
+     * Since C-strings are NULL-terminated, len must be +1
```

Comment:
The problem below is actually [H5Tset_size](https://hdfgroup.org/HDF5/doc/RM/RM_H5T.html#Datatype-SetSize) and the way it it handles fixed length strings:

```
String or character datatypes: The size set for a string datatype should include space for the null-terminator character, otherwise it will not be stored on (or retrieved from) disk.
```

So if I say "you have to set explicitly the NULL terminator and also tell me about it" it might be easier for users to think of it than saying "ignore it, I handle the length" since that might suggest users they can skip the `\0`, which they cant (and this class also has no control over that).